### DISCUSSION TITLE
Adding file or folder locks for nextjs fast refresh

### DISCUSSION THREAD

**Discussion by harryzec** [vercel/next.js#44152] [Ideas]
### Describe the feature you'd like to request

Hi - I work on a large enterprise app (with a lot of legacy code) and have an issue we have been facing since we upgraded to next 11 or 12 where we get a fast refresh with a full rebuild every time we change pages. It would basically be impossible at this point to find the cause of why this happens as there is a lot of old architecture code in there. The main thing that sucks is all of our stores/states (apollo, redux, in memory singleton stores) get knocked out every page and it makes local development impossible. Would it be possible to implement a feature where we can lock certain files from fast refresh when it a full rebuild is decided ? then I can initiate all of our state/stores in one folder and import them in to the _app file etc without them getting taken out on every pages. I realize there are packages to store this stuff in sessionStorage but that is pretty janky to do for all of our stores (and not possible for some variables).

### Describe the solution you'd like

In the next.js config could we have a section to lock certain files from fast refresh? i'm not sure exactly but in the config file do like 
```

fastRefresh: 
      lock: ["/store"]
```

### Describe alternatives you've considered

Storing/persisting apollo cache (apollo-cache-persist) and redux in session storage. not possible for some of our singleton stores though because we store functions in there. Maybe migrating our stores to another repository as an option ? kind of janky also though

**Comment by bugzpodder** [comment]
See if this affects you https://github.com/vercel/next.js/issues/44347
